datumaro/plugins/wider_format.py

```diff
@@ -0,0 +1,115 @@
+
+# Copyright (C) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import os
+import os.path as osp
+import re
+
+from datumaro.components.converter import Converter
+from datumaro.components.extractor import (AnnotationType, Bbox, DatasetItem,
+    Importer, SourceExtractor)
+
+
+class WiderPath:
+    IMAGE_EXT = '.jpg'
+    ANNOTATIONS_DIR = 'wider_face_split'
+    IMAGES_DIR = 'images'
+    SUBSET_DIR = 'WIDER_'
+    BBOX_ATTRIBUTES = ['blur', 'expression', 'illumination',
+        'occlusion', 'pose', 'invalid']
+
+class WiderExtractor(SourceExtractor):
+    def __init__(self, path):
+        assert osp.isfile(path), path
+        self._path = path
+        self._dataset_dir = osp.dirname(osp.dirname(path))
+
+        subset = osp.splitext(osp.basename(path))[0]
+        match = re.fullmatch(r'wider_face_\S+_bbx_gt', subset)
+        if match:
+            subset = subset.split('_')[2]
+        super().__init__(subset=subset)
+
+        self._items = list(self._load_items(path).values())
+
+    def _load_items(self, path):
+        items = {}
+        with open(path, 'r') as f:
+            lines = f.readlines()
+
+        image_ids = [image_id for image_id, line in enumerate(lines)
+            if WiderPath.IMAGE_EXT in line]
+
+        for image_id in image_ids:
+            image = lines[image_id]
+            image_path = osp.join(self._dataset_dir, WiderPath.SUBSET_DIR
+                + self._subset, WiderPath.IMAGES_DIR, image[:-1])
+            item_id = image[:-(len(WiderPath.IMAGE_EXT) + 1)]
+
+            bbox_count = lines[image_id + 1]
+            bbox_lines = lines[image_id + 2:image_id + 2 + int(bbox_count)]
```

Comment:
Please, maintain the formatting:
- spaces in slices
- ascending direction in comparisons (<, <=)